docs/index.bs

```diff
@@ -3175,6 +3283,103 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
+      1. Return the result of [=building a URLPattern from a Web IDL value=] |rawPattern| given |baseURL| and |serviceWorker|'s [=service worker/global object=]'s [=relevant realm=].
+  </section>
+
+  <section algorithm>
+    <h3 id="verify-router-rule-algorithm"><dfn>Verify Router Condition</dfn></h3>
+
+      : Input
+      :: |condition|, a {{RouterCondition}}
+      :: |serviceWorker|, a [=/service worker=]
+      : Output
+      :: a boolean
+
+      1. Let |hasCondition| be false.
+      1. If |condition|["{{RouterCondition/urlPattern}}"] [=map/exists=], then:
+          1. Let |rawPattern| be |condition|["{{RouterCondition/urlPattern}}"].
+          1. Let |pattern| be the result of running <a>Parse URL Pattern</a> algorithm passing |rawPattern| and |serviceWorker|. If this throws an exception, catch it and return false.
+          1. If |pattern| [=URLPattern/has regexp groups=], then return false.
+
+              Note: Since running a user-defined regular expression has a security concern, it is prohibited.
+
+          1. Set |hasCondition| to true.
+      1. If |condition|["{{RouterCondition/requestMethod}}"] [=map/exists=], set |hasCondition| to true.
+      1. If |condition|["{{RouterCondition/requestMode}}"] [=map/exists=], set |hasCondition| to true.
+      1. If |condition|["{{RouterCondition/requestDestination}}"] [=map/exists=], set |hasCondition| to true.
+      1. If |condition|["{{RouterCondition/runningStatus}}"] [=map/exists=], set |hasCondition| to true.
+      1. If |condition|["{{RouterCondition/_or}}"] [=map/exists=], then:
+          1. If |hasCondition| is true, return false.
+
+              Note: For ease of understanding the router rule, the "or" condition is mutually exclusive with other conditions.
+
+          1. Let |orConditions| be |condition|["{{RouterCondition/_or}}"].
+          1. For each |orCondition| of |orConditions|:
+              1. If running [=Verify Router Condition=] algorithm with |orCondition| and |serviceWorker| returns false, return false.
+          1. Set |hasCondition| to true.
+      1. Return |hasCondition|.
+  </section>
+
+  <section algorithm>
+    <h3 id="match-router-condition-algorithm"><dfn>Match Router Condition</dfn></h3>
+      : Input
+      :: |condition|, a {{RouterCondition}}
+      :: |serviceWorker|, a [=/service worker=]
+      :: |request|, a [=/request=]
+      : Output
+      :: a boolean
+
+      Note: if there are multiple conditions (e.g. `urlPattern`, `runningStatus`, and `requestMethod` are set), all conditions will be matched to return true.
+
+      1. If |condition|["{{RouterCondition/urlPattern}}"] [=map/exists=], then:
+          1. Let |rawPattern| be |condition|["{{RouterCondition/urlPattern}}"].
+          1. Let |pattern| be the result of running <a>Parse URL Pattern</a> algorithm passing |rawPattern| and |serviceWorker|.
+          1. If running [=match=] with |pattern| and |request|'s [=request/URL=] returns null, return false.
```

Comment:
Practically, URLPattern exec() and test() works because URL has necessary field to run [=match=].  However, I understand that you mean the type mismatch.

I have already contacted with @sisidovski, who is one of the authors of the URLPattern specification, to ask if we can make [=match=] accept URL.

Reply:
I also have filed https://github.com/whatwg/urlpattern/issues/218.